app/collections/collections-manager.jsx

```diff
@@ -1,46 +1,80 @@
 import React from 'react';
 import CollectionsCreateForm from './create-form';
 import CollectionSearch from './collection-search';
+import LoadingIndicator from '../components/loading-indicator';
 
 class CollectionsManager extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
+      adding: false,
       collections: [],
-      error: null,
-      hasCollectionSelected: false
+      errors: [],
+      hasCollectionSelected: true
     };
 
     this.addToCollections = this.addToCollections.bind(this);
+    this.onChange = this.onChange.bind(this);
+  }
+
+  onChange() {
+    const collections = this.search.getSelected();
+
+    if (collections.length > 0) {
+      this.setState({ hasCollectionSelected: false });
+    }
   }
 
   addToCollections() {
+    this.setState({ adding: true, errors: [] });
     const collections = this.search.getSelected();
-    if (!(collections.length > 0)) return;
+    if (collections.length === 0) return;
 
-    const promises = collections.map(collection =>
-      collection.collection.addLink('subjects', this.props.subjectIDs)
-    );
+    const promises = collections.map((searchResult) => {
+      const collection = searchResult.collection;
+      const subjectsToAdd = this.props.subjectIDs.filter((id) => {
+        if (!collection.links.subjects.includes(id)) return id;
```

Comment:
I'm not sure if I'm reading this correctly, but  the callback in `Array.filter()` should return a boolean for each array item, but this returns the item.

I think you mean to return `!collection.links.subjects.includes(id)`.